src/tabs/tab-header-bar.tsx

```diff
@@ -292,16 +343,36 @@ export function TabHeaderBar({
       [styles['tabs-tab-disabled']]: tab.disabled,
     });
 
+    const tabHeaderContainerClasses = clsx({
+      [styles['tabs-tab-header-container']]: true,
+      [styles.refresh]: isVisualRefresh,
+      [styles['tabs-tab-active']]: activeTabId === tab.id && !tab.disabled,
+      [styles['tabs-tab-disabled']]: tab.disabled,
+    });
+
     const commonProps: (JSX.IntrinsicElements['a'] | JSX.IntrinsicElements['button']) & { 'data-testid': string } = {
       className: classes,
-      role: 'tab',
-      'aria-selected': tab.id === activeTabId,
+      role: hasActionOrDismissible ? undefined : 'tab',
```

Comment:
why role is undefined for hasActionOrDismissible?

Reply:
This might be moreso a question regarding the implementation but the motivation was this: https://quip-amazon.com/GeTZAiyGeGYS/Tabs-with-Actions#temp:C:EYX9a272942d788438abdc7084b5

I looked into using `role='none` but since it's a synonym for the presentaton role, I didn't think either were a good fit. We do want the button to be interactable but we don't want to have an explicit role for it.

I'm open to suggestion though of course

Reply:
This may be a question moreso on the implementation but the A11y review advised us to remove any explicit role on this button. I can see if I can change this conditional to something like `!hasActionOrDismissible && role: 'tab'`